app/volunteers/dashboard/events/page.tsx

```diff
@@ -1,3 +1,90 @@
-export default function Events() {
-  return <>List of events</>;
+"use client";
+
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardFooter,
+  CardHeader,
+  CardTitle,
+} from "@/components/ui/card";
+import UsersWhite from "@/public/usersWhite";
+import CalendarWhite from "@/public/calendar_white";
+import { Button } from "@/components/ui/button";
+import useDashboardQuery from "@/hooks/use-dashboard-query";
+import Link from "next/link";
+import useSingleUserQuery from "@/hooks/use-single-user-query";
+import { UUID } from "crypto";
+import useRegisteredEventsQuery from "@/hooks/use-registered-events-query";
+
+export default function Dashboard() {
+  // Fetch data from database
+  const { data: events, status } = useDashboardQuery();
+
+  // Get user_data
+  const { data: user } = useSingleUserQuery();
+
+  const user_id = user?.volunteer_id;
+
+  // Fetch only the events that the user registered.
+  const { data: registeredEvents } = useRegisteredEventsQuery(user_id as UUID);
+
+  // If fetching is successful
+  if (status === "pending")
+    return <p className="text-accent text-2lg font-bold">Loading...</p>; // Will replace with skeleton
+
+  // Iterate all the row in the database = 'events' object.
+  const eventInfo = registeredEvents?.events?.map((event) => ({
```

Comment:
I'm not sure you need to do the 2 following `map`s, you can just access these by key sa imong `map` sa jsx.

Instead of:

```tsx
 </CardContent>
          {eventStatus?.map((status) => (
            <CardDescription>{status.status}</CardDescription>
          ))}
          <CardFooter>
```

You can do:

```tsx
 </CardContent>
          {registeredEvents?.map({ status } => (
            <CardDescription>{status}</CardDescription>
          ))}
          <CardFooter>
```

> Not sure though kay sa github rako ga edit, di nako makitang object types

Reply:
registeredEvents kay ang mga keys niya kay from event na table. Si eventStatus kay sa event_volunteer na table. Pero pwede ata ko mugamit ug spread syntax sa hook para included na sha sa registeredEvents. Will push later.

Reply:
Wrong diay, murag sakto rajud diay na. Irename nalang nako ng registered events to avoid confusion ug naming conflicts in the future.